### PR TITLE
ActiveDataProvider doc

### PR DESCRIPTION
**Pull request by fcaldarelli**

### CODE REVIEW

File: src/data/ActiveDataProvider.php [hunk 1]
```diff
@@ -23,30 +23,34 @@
...
  * ```php
  * $query = new Query();
- * $provider = new ActiveDataProvider([
- *     'query' => $query->from('post'),
- *     'pagination' => [
- *         'pageSize' => 20,
- *     ],
- * ]);
+ * $provider = new ActiveDataProvider(
+ *     Yii::$app->db,
```
Comment (hiqsol) [inline src/data/ActiveDataProvider.php]: `Yii::$app` is not available anymore.
Use `Yii::getApp()->get('db')` or in this case better simply `Yii::get('db')`

File: src/data/ActiveDataProvider.php [hunk 2]
```diff
@@ -13,7 +13,7 @@
 use yii\data\BaseDataProvider;
 use yii\db\Connection;
 use yii\db\QueryInterface;
-use yii\di\Instance;
+use yii\helpers\Yii;
```
Comment (hiqsol) [inline src/data/ActiveDataProvider.php]: AFAICS `Yii` is not used in this class.

File: src/data/ActiveDataProvider.php [hunk 3]
```diff
@@ -82,9 +86,10 @@ class ActiveDataProvider extends BaseDataProvider
 
 
     /**
-     * Initializes the DB connection component.
+     * Create the ActiveDataProvider object.
      * This method will initialize the [[db]] property to make sure it refers to a valid DB connection.
```
Comment (hiqsol) [inline src/data/ActiveDataProvider.php]: No need for such description. I think `@param` is enough in this case.

File: src/data/ActiveDataProvider.php (hunk 2 above)
Comment (samdark) [reply src/data/ActiveDataProvider.php]: https://github.com/yiisoft/active-record/commit/e93f97cd0cee1f23e217d4d32b2a88f8fab52b3a

File: src/data/ActiveDataProvider.php (hunk 1 above)
Comment (samdark) [reply src/data/ActiveDataProvider.php]: https://github.com/yiisoft/active-record/commit/e93f97cd0cee1f23e217d4d32b2a88f8fab52b3a

File: src/data/ActiveDataProvider.php (hunk 3 above)
Comment (samdark) [reply src/data/ActiveDataProvider.php]: https://github.com/yiisoft/active-record/commit/e93f97cd0cee1f23e217d4d32b2a88f8fab52b3a

**Pull request merged**